lib/tools/adb-commands.js

```diff
@@ -86,11 +86,14 @@ methods.getApiLevel = async function getApiLevel () {
       const strOutput = await this.getDeviceProperty('ro.build.version.sdk');
       let apiLevel = parseInt(strOutput.trim(), 10);
 
-      // Temp workaround. Android Q beta emulators report SDK 28 when they should be 29
-      if (apiLevel === 28 && (await this.getDeviceProperty('ro.build.version.release')).toLowerCase() === 'q') {
-        log.debug('Release version is Q but found API Level 28. Setting API Level to 29');
-        apiLevel = 29;
+      // Workaround for preview/beta platform API level
+      const asciiq = 113; // 113 is the code point of 'q'
+      const apiLevelDiff = apiLevel - 28; // 28 is the first API Level getPlatformVersion() returns alphabet
```

Comment:
... is the first API Level, where Android SDK started returning letters in response to `getPlatformVersion`